from PyQt5 import QtCore, QtGui, QtWidgets
from addbankname import*
from adminlogin import*
from agentdetails import*
from Agentregistrationpage import*
from agentsearch import*
# from all_details import*
from bank_details import*
from changePassword import*
from declaration2 import*
from EMIINTEREST import*
from loan_request import*
from loantype import*
from registrationpage import*
from requestpage import*
from searching import*
from reportApprove import *
from reportmonthlyapprove import *
from customerdetails import *
from bank_details import*

class Ui_dashBoard(object):
    def __init__(self,e):
        self.mail = e
        
    def setupUi(self, dashBoard):
        dashBoard.setObjectName("dashBoard")
        dashBoard.resize(1368, 878)
        dashBoard.setMinimumSize(QtCore.QSize(1368, 878))
        dashBoard.setMaximumSize(QtCore.QSize(1368, 878))
        dashBoard.setStyleSheet("background:rgb(183, 246, 255)")
        self.centralwidget = QtWidgets.QWidget(dashBoard)
        self.centralwidget.setObjectName("centralwidget")
        self.horizontalLayout = QtWidgets.QHBoxLayout(self.centralwidget)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.frame = QtWidgets.QFrame(self.centralwidget)
        self.frame.setStyleSheet("background:rgb(183, 246, 255)")
        self.frame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.frame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.frame.setObjectName("frame")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.frame)
        self.verticalLayout.setObjectName("verticalLayout")
        self.label = QtWidgets.QLabel(self.frame)
        self.label.setMaximumSize(QtCore.QSize(16777215, 80))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(22)
        font.setBold(False)
        font.setWeight(50)
        self.label.setFont(font)
        self.label.setStyleSheet("background:rgb(183, 246, 255)\n"
"")
        self.label.setObjectName("label")
        self.verticalLayout.addWidget(self.label)
        self.b1 = QtWidgets.QPushButton(self.frame)
        self.b1.setMaximumSize(QtCore.QSize(16777215, 50))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.b1.setFont(font)
        self.b1.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.b1.setStyleSheet("background:light silver")
        self.b1.setObjectName("b1")
        self.verticalLayout.addWidget(self.b1)
        self.b2 = QtWidgets.QPushButton(self.frame)
        self.b2.setMaximumSize(QtCore.QSize(16777215, 50))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.b2.setFont(font)
        self.b2.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.b2.setStyleSheet("background:light silver\n"
"")
        self.b2.setObjectName("b2")
        self.verticalLayout.addWidget(self.b2)
        self.b3 = QtWidgets.QPushButton(self.frame)
        self.b3.setMaximumSize(QtCore.QSize(16777215, 50))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.b3.setFont(font)
        self.b3.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.b3.setStyleSheet("background:light silver")
        self.b3.setObjectName("b3")
        self.verticalLayout.addWidget(self.b3)
        self.b4 = QtWidgets.QPushButton(self.frame)
        self.b4.setMaximumSize(QtCore.QSize(16777215, 50))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.b4.setFont(font)
        self.b4.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.b4.setStyleSheet("background:light silver")
        self.b4.setObjectName("b4")
        self.verticalLayout.addWidget(self.b4)
        self.horizontalLayout.addWidget(self.frame, 0, QtCore.Qt.AlignHCenter)
        dashBoard.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(dashBoard)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1368, 26))
        self.menubar.setObjectName("menubar")
        self.menuAdd_Items = QtWidgets.QMenu(self.menubar)
        self.menuAdd_Items.setObjectName("menuAdd_Items")
        self.menuReport = QtWidgets.QMenu(self.menubar)
        self.menuReport.setObjectName("menuReport")
        self.menuReturn = QtWidgets.QMenu(self.menuReport)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.menuReturn.setFont(font)
        self.menuReturn.setObjectName("menuReturn")
        self.menuSettings = QtWidgets.QMenu(self.menubar)
        self.menuSettings.setObjectName("menuSettings")
        self.menuSEARCHING = QtWidgets.QMenu(self.menubar)
        self.menuSEARCHING.setObjectName("menuSEARCHING")
        self.menuVIEW = QtWidgets.QMenu(self.menubar)
        self.menuVIEW.setObjectName("menuVIEW")
        dashBoard.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(dashBoard)
        self.statusbar.setObjectName("statusbar")
        dashBoard.setStatusBar(self.statusbar)
        self.actionSubject = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionSubject.setFont(font)
        self.actionSubject.setObjectName("actionSubject")
        self.actionCourse = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionCourse.setFont(font)
        self.actionCourse.setObjectName("actionCourse")
        self.actionDepartment = QtWidgets.QAction(dashBoard)
        self.actionDepartment.setObjectName("actionDepartment")
        self.actionAdd = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionAdd.setFont(font)
        self.actionAdd.setObjectName("actionAdd")
        self.actionPurchase = QtWidgets.QAction(dashBoard)
        self.actionPurchase.setObjectName("actionPurchase")
        self.actionAll_purchase = QtWidgets.QAction(dashBoard)
        self.actionAll_purchase.setObjectName("actionAll_purchase")
        self.actionMonthly_purchase = QtWidgets.QAction(dashBoard)
        self.actionMonthly_purchase.setObjectName("actionMonthly_purchase")
        self.actionAll_issue = QtWidgets.QAction(dashBoard)
        self.actionAll_issue.setObjectName("actionAll_issue")
        self.actionMonthly_issue = QtWidgets.QAction(dashBoard)
        self.actionMonthly_issue.setObjectName("actionMonthly_issue")
        self.actionAll_return = QtWidgets.QAction(dashBoard)
        self.actionAll_return.setObjectName("actionAll_return")
        self.actionMonthly_return = QtWidgets.QAction(dashBoard)
        self.actionMonthly_return.setObjectName("actionMonthly_return")
        self.actionAll_fine = QtWidgets.QAction(dashBoard)
        self.actionAll_fine.setObjectName("actionAll_fine")
        self.actionMonthly_fine = QtWidgets.QAction(dashBoard)
        self.actionMonthly_fine.setObjectName("actionMonthly_fine")
        self.actionStudent_Report = QtWidgets.QAction(dashBoard)
        self.actionStudent_Report.setObjectName("actionStudent_Report")
        self.actionBook_Stock = QtWidgets.QAction(dashBoard)
        self.actionBook_Stock.setObjectName("actionBook_Stock")
        self.actionChange_Password = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionChange_Password.setFont(font)
        self.actionChange_Password.setObjectName("actionChange_Password")
        self.actionProfile = QtWidgets.QAction(dashBoard)
        self.actionProfile.setObjectName("actionProfile")
        self.actionLOAN_REQUEST = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionLOAN_REQUEST.setFont(font)
        self.actionLOAN_REQUEST.setObjectName("actionLOAN_REQUEST")
        self.actionLOAN_REQUEST_PAGE = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionLOAN_REQUEST_PAGE.setFont(font)
        self.actionLOAN_REQUEST_PAGE.setObjectName("actionLOAN_REQUEST_PAGE")
        self.actionCUSTOMER_DETAILS = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionCUSTOMER_DETAILS.setFont(font)
        self.actionCUSTOMER_DETAILS.setObjectName("actionCUSTOMER_DETAILS")
        self.actionAGENT_DETAILS = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionAGENT_DETAILS.setFont(font)
        self.actionAGENT_DETAILS.setObjectName("actionAGENT_DETAILS")
        self.actionCUSTOMER_SEARCHING = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionCUSTOMER_SEARCHING.setFont(font)
        self.actionCUSTOMER_SEARCHING.setObjectName("actionCUSTOMER_SEARCHING")
        self.actionAGENT_SEARCHING = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionAGENT_SEARCHING.setFont(font)
        self.actionAGENT_SEARCHING.setObjectName("actionAGENT_SEARCHING")
        self.actionALL_DETAILS = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionALL_DETAILS.setFont(font)
        self.actionALL_DETAILS.setObjectName("actionALL_DETAILS")
        self.actionBANK_DETAILS = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionBANK_DETAILS.setFont(font)
        self.actionBANK_DETAILS.setObjectName("actionBANK_DETAILS")
        self.actionBANK_DETAILS_2 = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionBANK_DETAILS_2.setFont(font)
        self.actionBANK_DETAILS_2.setObjectName("actionBANK_DETAILS_2")
        self.actionEMI_CALCULATOR = QtWidgets.QAction(dashBoard)
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.actionEMI_CALCULATOR.setFont(font)
        self.actionEMI_CALCULATOR.setObjectName("actionEMI_CALCULATOR")
        self.menuAdd_Items.addAction(self.actionSubject)
        self.menuAdd_Items.addAction(self.actionCourse)
        self.menuReturn.addAction(self.actionAll_return)
        self.menuReturn.addAction(self.actionMonthly_return)
        self.menuReport.addAction(self.menuReturn.menuAction())
        self.menuReport.addAction(self.actionCUSTOMER_DETAILS)
        self.menuReport.addAction(self.actionAGENT_DETAILS)
        self.menuReport.addAction(self.actionALL_DETAILS)
        self.menuSettings.addAction(self.actionChange_Password)
        self.menuSEARCHING.addAction(self.actionCUSTOMER_SEARCHING)
        self.menuSEARCHING.addAction(self.actionAGENT_SEARCHING)
        self.menuVIEW.addAction(self.actionBANK_DETAILS_2)
        self.menuVIEW.addAction(self.actionEMI_CALCULATOR)
        self.menubar.addAction(self.menuAdd_Items.menuAction())
        self.menubar.addAction(self.menuVIEW.menuAction())
        self.menubar.addAction(self.menuReport.menuAction())
        self.menubar.addAction(self.menuSEARCHING.menuAction())
        self.menubar.addAction(self.menuSettings.menuAction())

        
        self.b1.clicked.connect(self.addcustomer)
        self.b2.clicked.connect(self.addagent)
        self.b3.clicked.connect(self.loanrequest)
        self.b4.clicked.connect(self.listloanrequest)
        self.menubar.triggered[QtWidgets.QAction].connect(self.menu)

        self.retranslateUi(dashBoard)
        QtCore.QMetaObject.connectSlotsByName(dashBoard)

    def retranslateUi(self, dashBoard):
        _translate = QtCore.QCoreApplication.translate
        dashBoard.setWindowTitle(_translate("dashBoard", "Dashboard"))
        self.label.setText(_translate("dashBoard", "   Dashboard"))
        self.b1.setText(_translate("dashBoard", "Add customer"))
        self.b2.setText(_translate("dashBoard", "Add add agent"))
        self.b3.setText(_translate("dashBoard", "Loan Request"))
        self.b4.setText(_translate("dashBoard", "List of Loan requests"))
        self.menuAdd_Items.setTitle(_translate("dashBoard", "ADD ITEMS"))
        self.menuReport.setTitle(_translate("dashBoard", "REPORT"))
        self.menuReturn.setTitle(_translate("dashBoard", "APPROVE DETAILS"))
        self.menuSettings.setTitle(_translate("dashBoard", "SETTINGS"))
        self.menuSEARCHING.setTitle(_translate("dashBoard", "SEARCHING"))
        self.menuVIEW.setTitle(_translate("dashBoard", "VIEW"))
        self.actionSubject.setText(_translate("dashBoard", "ADD BANK NAME"))
        self.actionCourse.setText(_translate("dashBoard", "ADD LOAN TYPE"))
        self.actionDepartment.setText(_translate("dashBoard", "Department"))
        self.actionAdd.setText(_translate("dashBoard", "LOAN REQUEST"))
        self.actionPurchase.setText(_translate("dashBoard", "Purchase"))
        self.actionAll_purchase.setText(_translate("dashBoard", "All purchase"))
        self.actionMonthly_purchase.setText(_translate("dashBoard", "Monthly purchase"))
        self.actionAll_issue.setText(_translate("dashBoard", "All issue"))
        self.actionMonthly_issue.setText(_translate("dashBoard", "Monthly issue"))
        self.actionAll_return.setText(_translate("dashBoard", "ApproveDetails"))
        self.actionMonthly_return.setText(_translate("dashBoard", "MONTHLY DETAILS"))
        self.actionAll_fine.setText(_translate("dashBoard", "All fine"))
        self.actionMonthly_fine.setText(_translate("dashBoard", "Monthly fine"))
        self.actionStudent_Report.setText(_translate("dashBoard", "Student Report"))
        self.actionBook_Stock.setText(_translate("dashBoard", "Book Stock"))
        self.actionChange_Password.setText(_translate("dashBoard", "CHANGE PASSWORD"))
        self.actionProfile.setText(_translate("dashBoard", "Profile"))
        self.actionLOAN_REQUEST.setText(_translate("dashBoard", "LOAN REQUEST"))
        self.actionLOAN_REQUEST_PAGE.setText(_translate("dashBoard", "LOAN REQUEST PAGE"))
        self.actionCUSTOMER_DETAILS.setText(_translate("dashBoard", "CUSTOMER DETAILS"))
        self.actionAGENT_DETAILS.setText(_translate("dashBoard", "AGENT DETAILS"))
        self.actionCUSTOMER_SEARCHING.setText(_translate("dashBoard", "CUSTOMER SEARCHING"))
        self.actionAGENT_SEARCHING.setText(_translate("dashBoard", "AGENT SEARCHING"))
        # self.actionALL_DETAILS.setText(_translate("dashBoard", "ALL DETAILS"))
        self.actionBANK_DETAILS.setText(_translate("dashBoard", "BANK DETAILS"))
        self.actionBANK_DETAILS_2.setText(_translate("dashBoard", "BANK DETAILS"))
        self.actionEMI_CALCULATOR.setText(_translate("dashBoard", "EMI CALCULATOR"))

    def menu(self,action):
        txt = (action.text())
        
        if txt == "ADD BANK NAME":
            self.Add_Bank_Name = QtWidgets.QMainWindow()
            self.ui = Ui_Add_Bank_Name()
            self.ui.setupUi(self.Add_Bank_Name)
            self.Add_Bank_Name.show()
        if txt == "ADD LOAN TYPE":
            self.LoanType = QtWidgets.QMainWindow()
            self.ui = Ui_LoanType()
            self.ui.setupUi(self.LoanType)
            self.LoanType.show()

        if txt == "CUSTOMER SEARCHING":
            self.CSP = QtWidgets.QMainWindow()
            self.ui = Ui_CSP()
            self.ui.setupUi(self.CSP)
            self.CSP.show()
        
        if txt == "AGENT SEARCHING":
            self.agentsearch = QtWidgets.QMainWindow()
            self.ui = Ui_agentsearch()
            self.ui.setupUi(self.agentsearch)
            self.agentsearch.show()

        if txt == "CUSTOMER DETAILS":
            self.CUSTOMERDETAILS = QtWidgets.QMainWindow()
            self.ui = Ui_CUSTOMERDETAILS()
            self.ui.setupUi(self.CUSTOMERDETAILS)
            self.CUSTOMERDETAILS.show()

        if txt == "ApproveDetails":
            self.repapp = QtWidgets.QMainWindow()
            self.ui = Ui_repapp()
            self.ui.setupUi(self.repapp)
            self.repapp.show()

            # msg = QMessageBox()
            # msg.setIcon(QMessageBox.Critical)
            # msg.setText("Purchase not done yet")
            # msg.setWindowTitle("Invalid")
            # msg.show()
            # msg.exec_() 
        
        if txt == "MONTHLY DETAILS":
            self.repappMon = QtWidgets.QMainWindow()
            self.ui = Ui_repappMon()
            self.ui.setupUi(self.repappMon)
            self.repappMon.show()

        # if txt == "ALL DETAILS":
        #     self.Coualldetail = QtWidgets.QMainWindow()
        #     self.ui = Ui_Coualldetail()
        #     self.ui.setupUi(self.Coualldetail)
        #     self.Coualldetail.show()

        if txt == "AGENT DETAILS":
            self.Agentdet = QtWidgets.QMainWindow()
            self.ui = Ui_Agentdet()
            self.ui.setupUi(self.Agentdet)
            self.Agentdet.show()
        
        if txt == "CHANGE PASSWORD":
            self.changePa = QtWidgets.QMainWindow()
            self.ui = Ui_changePa(self.mail)
            self.ui.setupUi(self.changePa)
            self.changePa.show()

        if txt =="BANK DETAILS":
            self.BANK_DETAILS = QtWidgets.QMainWindow()
            self.ui = Ui_BANK_DETAILS()
            self.ui.setupUi(self.BANK_DETAILS)
            self.BANK_DETAILS.show()
            
        if txt =="EMI CALCULATOR":
            self.emipage = QtWidgets.QMainWindow()
            self.ui = Ui_emipage()
            self.ui.setupUi(self.emipage)
            self.emipage.show()


    def addcustomer(self):
        self.cReg = QtWidgets.QMainWindow()
        self.ui = Ui_cReg()
        self.ui.setupUi(self.cReg)
        self.cReg.show()

    def addagent(self):
        self.Registration_page = QtWidgets.QMainWindow()
        self.ui = Ui_Registration_page()
        self.ui.setupUi(self.Registration_page)
        self.Registration_page.show()

    def loanrequest(self):
        self.loan_request = QtWidgets.QMainWindow()
        self.ui = Ui_loan_request()
        self.ui.setupUi(self.loan_request)
        self.loan_request.show()

    def listloanrequest(self):
        self.MainWindow = QtWidgets.QMainWindow()
        self.ui = Ui_MainWindow()
        self.ui.setupUi(self.MainWindow)
        self.MainWindow.show()


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    dashBoard = QtWidgets.QMainWindow()
    ui = Ui_dashBoard()
    ui.setupUi(dashBoard)
    dashBoard.show()
    sys.exit(app.exec_())
